src/OrchardCore/OrchardCore.Users.Core/Authentication/CacheTicketStore.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.IO;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.Users.Authentication;
+
+public class CacheTicketStore : ITicketStore
+{
+    private const string _keyPrefix = "ocauth-ticket";
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private IDataProtector _dataProtector;
+
+    public CacheTicketStore(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    public async Task RemoveAsync(string key)
+    {
+        var cacheKey = $"{_keyPrefix}-{key}";
+        var cache = _httpContextAccessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        await cache.RemoveAsync(cacheKey);
+    }
+
+    public async Task RenewAsync(string key, AuthenticationTicket ticket)
+    {
+        var cacheKey = $"{_keyPrefix}-{key}";
+        var cache = _httpContextAccessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        _dataProtector ??= _httpContextAccessor.HttpContext.RequestServices.GetService<IDataProtectionProvider>()
+                                .CreateProtector($"{nameof(CacheTicketStore)}_{IdentityConstants.ApplicationScheme}");
+
+        var userId = string.Empty;
+        var userName = string.Empty;
+        if (ticket.AuthenticationScheme == IdentityConstants.ApplicationScheme)
+        {
+            userId = ticket.Principal.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+            userName = ticket.Principal.FindFirst(ClaimTypes.Name)?.Value ?? ticket.Principal.FindFirst("name")?.Value;
+        }
+
+        var protectedBytes = _dataProtector.Protect(SerializeTicket(ticket));
+        var data = new TicketRecord
+        {
+            Key = key,
+            UserId = userId,
+            Username = userName,
+            LastActivity = DateTime.UtcNow,
+            Value = protectedBytes,
+            Expires = ticket.Properties.ExpiresUtc.Value.UtcDateTime,
+        };
+
+        var bytes = Serialize(data);
+        await cache.SetAsync(cacheKey, bytes, new DistributedCacheEntryOptions() { AbsoluteExpiration = ticket.Properties.ExpiresUtc.Value });
+    }
+    public async Task<AuthenticationTicket> RetrieveAsync(string key)
+    {
+        var cacheKey = $"{_keyPrefix}-{key}";
+        var cache = _httpContextAccessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        var bytes = await cache.GetAsync(cacheKey);
+        if (bytes == null || bytes.Length == 0)
+            return null;
```

Comment:
```suggestion
        {
            return null;
        }
```

Reply:
I guess single line is fine with without `{` and `}`